freight/api/task_index.py

```diff
@@ -98,6 +78,7 @@ def get(self):
     post_parser = reqparse.RequestParser()
     post_parser.add_argument('app', required=True)
     post_parser.add_argument('user', required=True)
+    post_parser.add_argument('name', default=TaskName.deploy)
```

Comment:
We actually had been talking internally about killing off the name attribute as its kind of useless (we only added it because Heaven had it).

One thing we did want to do have the ability to specify arbitrary params which could be passed down to providers. In that case if you really needed a dynamic task name, that could be one of those params. Whats your thoughts on that?

i.e. it'd be something like 'params = {...}' to the endpoint, and those would get passed into the formatter so they could be used in whatever way is needed